modules/apps/frontend-data-set/frontend-data-set-impl/src/main/java/com/liferay/frontend/data/set/internal/serializer/SystemFDSSerializer.java

```diff
@@ -375,6 +375,10 @@ private JSONArray _serializeCustomViews(
 			(ThemeDisplay)httpServletRequest.getAttribute(
 				WebKeys.THEME_DISPLAY);
 
+		String filterExpression = StringBundler.concat(
+			"'fdsName' eq '", fdsName, "' and 'creatorId' eq '",
+			themeDisplay.getUserId(), "'");
+
```

Comment:
if `_serializeCustomViews()` has the same logic in both serializers, then we need to hoist up this method to a the parent `BaseFDSSerializer` abstract class.

Method in the Base class should be `protected` and be named `serializeCustomViews` (w/o the `_` as it's no longer private).

You'll need also to hoist up `_objectDefinitionLocalService` into a protected attribute.

You can use what's already in the Base class as reference. We can checkup together in friday if needed